drivers/flash/flash_stm32_ex_op.c

```diff
@@ -85,13 +88,79 @@ int flash_stm32_ex_op_sector_wp(const struct device *dev, const uintptr_t in,
 #endif /* CONFIG_FLASH_STM32_WRITE_PROTECT */
 
 #if defined(CONFIG_FLASH_STM32_READOUT_PROTECTION)
+int flash_stm32_ex_op_update_rdp(const struct device *dev, bool enable,
+			   bool permanent)
+{
+	uint8_t current_level, target_level;
+
+	current_level = flash_stm32_get_rdp_level(dev);
+	target_level = current_level;
+
+	/*
+	 * 0xAA = RDP level 0 (no protection)
+	 * 0xCC = RDP level 2 (permanent protection)
+	 * others = RDP level 1 (protection active)
+	 */
+	switch (current_level) {
+	case FLASH_STM32_RDP2:
+		if (!enable || !permanent) {
+			LOG_ERR("RDP level 2 is permanent and can't be changed!");
+			return -ENOTSUP;
+		}
+		break;
+	case FLASH_STM32_RDP0:
+		if (enable) {
+			target_level = FLASH_STM32_RDP1;
+			if (permanent) {
+#if defined(CONFIG_FLASH_STM32_READOUT_PROTECTION_PERMANENT_ALLOW)
+				target_level = FLASH_STM32_RDP2;
+#else
+				LOG_ERR("Permanent readout protection (RDP "
+					"level 0 -> 2) not allowed");
+				return -ENOTSUP;
+#endif
```

Comment:
Not my call here, that is up to STM mainteiner, but wouldn't it be better to have the message as LOG_DBG?
The ERR messages are quite long and:
 1) I doubt they will be useful more than once, before tests figure out that code should not use the feature or the feature should be enabled;
 2) They will be always there, uselessly hanging there in rom.

Reply:
Looks like these were originally asserts, which were then changed into LOG_ERR:
https://github.com/zephyrproject-rtos/zephyr/commit/e7799eb55187f30c8755b63606f42dac937b2f77#diff-f7e9be53ed802bc0c668dc70d020bccd2ea03cba0217792ec471948f27efe336

Personally, I would have also chosen asserts. But I've got no real strong opinion on this, both options are fine by me.

Reply:
Agreed to move to LOG_DBG. Thanks for pointing this.

Reply:
done